crates/omnix-cli/crate.nix

```diff
@@ -39,10 +39,14 @@ in
           name = "flakreate-registry";
           src = flake.inputs.self + /crates/flakreate/registry;
         };
+      DEFAULT_FLAKE_SCHEMAS = lib.cleanSourceWith {
+        name = "flake-schemas";
+        src = flake.inputs.self + /nix/flake-schemas;
+      };
```

Comment:
I think this can be replaced with:

```
DEFAULT_FLAKE_SCHEMAS = flake.inputs.self + /nix/flake-schemas;
```

It should behave the same since there is no `filter` provided for `cleanSourceWith`?

Reply:
@srid why does `OM_INIT_REGISTRY` use `cleanSourceWith` if there is no `filter`?

Reply:
That's the actual semantics of `cleanSourceWith`, 

Proof:

```
❯ echo $OM_INIT_REGISTRY
/nix/store/9j2sil7h6w2lx7p71x2khhy0ba7v0i9d-flakreate-registry
```

With,

```diff
diff --git a/crates/omnix-cli/crate.nix b/crates/omnix-cli/crate.nix
index 0f99d6d..04043bd 100644
--- a/crates/omnix-cli/crate.nix
+++ b/crates/omnix-cli/crate.nix
@@ -34,11 +34,7 @@ in
         ) ++ lib.optionals pkgs.stdenv.isLinux [
         pkgsStatic.openssl
       ];
-      OM_INIT_REGISTRY =
-        lib.cleanSourceWith {
-          name = "flakreate-registry";
-          src = flake.inputs.self + /crates/flakreate/registry;
-        };
+      OM_INIT_REGISTRY = flake.inputs.self + /crates/flakreate/registry;
       DEVOUR_FLAKE = inputs.devour-flake;
       OMNIX_SOURCE = inputs.self;
       NIX_FLAKE_SCHEMAS_BIN = lib.getExe pkgs.nix-flake-schemas;
```

That becomes:

```
❯ echo $OM_INIT_REGISTRY
/nix/store/fsrav4xa7wrhnw19h04r4jk7cdpkbh6g-source/crates/flakreate/registry
```

@roberth was the one originally came across this pattern, in haskell-flake:

https://github.com/srid/haskell-flake/blob/e48718c26fa9ba2e7c2b3a8e5df2781b64acb011/nix/build-haskell-package.nix#L15-L24

Since this function is not self-documenting (what does "clean source" even mean here?) perhaps there should be another function that is?